pkg/liveshare/port_forwarder.go

```diff
@@ -54,6 +60,12 @@ func (fwd *PortForwarder) ForwardToListener(ctx context.Context, listen net.List
 		return err
 	}
 
+	// Ping the port to ensure that it is fully forwarded before continuing
+	err = WaitForPortConnection(ctx, listen.Addr().String())
+	if err != nil {
+		return err
+	}
```

Comment:
wdyt if we just let callers handle this? I worry about such a low-level change to an API that's been used for a long time. Callers worried about races can use the `WaitForPortConnection` method you are exporting.

Reply:
Makes sense to me! For the time being, I'll just use this for the gRPC client and we can update other areas that could use this going forward.